upgrade-tests/src/test/java/io/zeebe/test/UpgradeTest.java

```diff
@@ -111,17 +112,21 @@
             scenario()
                 .deployWorkflow(incidentWorkflow())
                 .createInstance()
-                .beforeUpgrade(UpgradeTest::createIncident)
+                .beforeUpgrade(UpgradeTest::awaitIncidentCreation)
                 .afterUpgrade(UpgradeTest::resolveIncident)
                 .done()
           },
           {
             "publish message",
             scenario()
                 .deployWorkflow(messageWorkflow())
-                .createInstance()
-                .beforeUpgrade(UpgradeTest::awaitOpenMessageSubscription)
-                .afterUpgrade(UpgradeTest::publishMessage)
+                .createInstance(Map.of("key", "123"))
+                .beforeUpgrade(
+                    state -> {
+                      publishMessage(state, -1L, -1L);
+                      return -1L;
+                    })
+                .afterUpgrade((state, k1, k2) -> awaitOpenMessageSubscription(state))
```

Comment:
Not exactly. We should create the instance after the upgrade and verify that the previously published message is correlated.